lib/spack/spack/main.py

```diff
@@ -343,17 +343,21 @@ def add_command(self, cmd_name):
                 self._remove_action(self._actions[-1])
             self.subparsers = self.add_subparsers(metavar="COMMAND", dest="command")
 
-        # each command module implements a parser() function, to which we
-        # pass its subparser for setup.
-        module = spack.cmd.get_module(cmd_name)
-
-        # build a list of aliases
-        alias_list = [k for k, v in aliases.items() if v == cmd_name]
-
-        subparser = self.subparsers.add_parser(
-            cmd_name, aliases=alias_list, help=module.description, description=module.description
-        )
-        module.setup_parser(subparser)
+        if cmd_name not in self.subparsers._name_parser_map:
```

Comment:
This is dangerous, cause we might not be forward compatible with Python 3.12+ whenever they feel like changing internals.

Reply:
True, but I didn't see ani API call to get the same information. We should probably revisit the code here, but not a week before release?

Reply:
How about the second best option of doing the bookkeeping with our own `set` somewhere? Then at least it's more likely that people can use an old commit of Spack with a new Python in the future. Best solution would of course be to prevent the same command from being attempted to be registered twice, but well...

Reply:
FWIW the internals of argparse (at least the field names) seldom change. We are already using internal fields above, so there’s more danger than just this. I don’t particularly *like* it, but this is all to support lazy command loading — which makes spack a lot snappier on a shared FS.

Where’s this getting called twice for the same command in spack? I could look at fixing that.

Reply:
> Where’s this getting called twice for the same command in spack? I could look at fixing that.

As far as I can tell it's only `spack help` code path that results in adding the same definition of `help` twice.